MagnetoSoftwareTesting/src/test/java/MagnetoTest/MagnetoApplicationTest.java

```diff
@@ -0,0 +1,164 @@
+package MagnetoTest;
+
+import java.util.concurrent.TimeUnit;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.interactions.Actions;
+import org.testng.Assert;
+import org.testng.annotations.AfterTest;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Parameters;
+import org.testng.annotations.Test;
+
+
+
+public class MagnetoApplicationTest {
+public String Url ="https://magento.softwaretestingboard.com";  
+//String driverpath ="C:\\Users\\rajashrees\\Desktop\\Seljars\\chromedriver_win32 (4)\\chromedriver.exe"; 
+//String driverName = "webdriver.chrome.driver";  
+public WebDriver driver; 
+
+@Parameters("browser")
+@BeforeTest  
+public void browser(String browser)  
+{	  
+//System.setProperty(driverName,driverpath);  
+//driver= new ChromeDriver();	
+Driver d=new Driver(browser);
+driver=d.setDriver();
+driver.manage().window().maximize();     
+driver.get(Url);  
+driver.manage().timeouts().implicitlyWait(5,TimeUnit.SECONDS);  
+}	  
+
+@Test(priority = 0)  
+public void chooseProduct() throws InterruptedException 
+{  
+    Actions act = new Actions(driver);
+    
+	WebElement Women =driver.findElement(By.xpath("//*[@id=\"ui-id-4\"]/span[2]"));
+	act.moveToElement(Women).perform();
+	Thread.sleep(1000);
```

Comment:
Use explicit wait. Remove Thread.sleep

Reply:
Use Explicit and implicit wait and removed Thread.sleep().